substrate/frame/preimage/src/lib.rs

```diff
@@ -195,15 +229,80 @@ pub mod pallet {
 		///
 		/// NOTE: THIS MUST NOT BE CALLED ON `hash` MORE TIMES THAN `request_preimage`.
 		#[pallet::call_index(3)]
-		#[pallet::weight(T::WeightInfo::unrequest_preimage())]
 		pub fn unrequest_preimage(origin: OriginFor<T>, hash: T::Hash) -> DispatchResult {
 			T::ManagerOrigin::ensure_origin(origin)?;
 			Self::do_unrequest_preimage(&hash)
 		}
+
+		/// Ensure that the a bulk of pre-images is upgraded.
+		///
+		/// The caller pays no fee if at least 90% of pre-images were successfully upgraded.
+		#[pallet::call_index(4)]
+		#[pallet::weight(T::WeightInfo::ensure_updated(hashes.len() as u32))]
+		pub fn ensure_updated(
+			origin: OriginFor<T>,
+			hashes: Vec<T::Hash>,
```

Comment:
We don't require calls be bounded and we don't know ahead of time exactly what this bound would be, plus it might change as we update benchmarks/weights.